### PR TITLE
[Mono] Add intrinsics for several operators

### PR DESCRIPTION
**Pull request by mkhamoyan**
This PR contributes to  #64072

Added intrinsics for following operators
- Operator *
- Operator /
- Operator |
- Operator &
- Operator ^
- Unary operator -
- Unary operator ~

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1055,15 +1060,57 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+				ins->inst_c0 = OP_FMUL;
+				break;
+			case SN_op_Division:
+				ins->inst_c0 = OP_FDIV;
+				break;
+			default:
+				NULLIFY_INS (ins);
+				return NULL;
+			}
+		} else {
```
Comment (simonrozsival) [inline src/mono/mono/mini/simd-intrinsics.c:1092]: You haven't implemented integer multiplication and division. Is there a reason for that? Will those be part of a different PR?

Comment (mkhamoyan) [reply src/mono/mono/mini/simd-intrinsics.c:1092]: In all places divide and division (method, operators) are not implemented for integer so I decided to do the same. In emit_sys_numerics_vector_t 'SN_op_Division' and 'SN_op_Multiply' are commented for integer so not sure if there is a reason for it or no.

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c:1092]: For `Division` method, I only found intrinsics for non-integer types. For `Multiply` method, there is intrinsics for all types.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -1055,15 +1060,57 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		break;
 	}
 	case SN_op_Addition:
-	case SN_op_Subtraction: {	
+	case SN_op_Subtraction:
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1063]: Let's order them alphabetically so it is easier to read and maintain.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -1055,15 +1060,57 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+			case SN_op_Addition:
+				ins->inst_c0 = OP_IADD;
+				break;
+			case SN_op_Subtraction:
+				ins->inst_c0 = OP_ISUB;
+				break;
+			case SN_op_BitwiseAnd:
+				ins->inst_c0 = OP_IAND;
+				break;
+			case SN_op_BitwiseOr:
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1103]: I just merged this PR (https://github.com/dotnet/runtime/pull/65486/files), where I adjusted the op code for and, or and xor. They now could handle both integer and floating point types. Please adjust them in this PR too.

File: src/mono/mono/mini/simd-intrinsics.c (hunk 1 above)
Comment (tannergooding) [reply src/mono/mono/mini/simd-intrinsics.c:1114]: Right, there is no direct SIMD acceleration support for integer division, just multiply.

You could accelerate a few cases of division by constant, but that's not something that has been a focus for the JIT yet.

**Comment by fanyang-mono** [comment]
@vargaz Could you please review again?

File: src/mono/mono/mini/simd-intrinsics.c [hunk 4]
```diff
@@ -1197,27 +1240,32 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
 	case SN_op_Addition:
-	case SN_op_Subtraction: {	
-		if (!(fsig->param_count == 2 && mono_metadata_type_equal (fsig->ret, type) && mono_metadata_type_equal (fsig->params [0], type) && mono_metadata_type_equal (fsig->params [1], type)))
+	case SN_op_BitwiseAnd:
+	case SN_op_BitwiseOr:
+	case SN_op_Division:
+	case SN_op_ExclusiveOr:
+	case SN_op_Multiply:
+	case SN_op_Subtraction: {
+		if (fsig->param_count != 2 )
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1249]: Will have a follow-up PR to fix the extra space.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 5]
```diff
@@ -1197,27 +1240,32 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+		MonoTypeEnum arg0_type = fsig->param_count > 0 ? get_underlying_type (fsig->params [0]) : MONO_TYPE_VOID;
+		return emit_simd_ins_for_binary_op (cfg, klass, fsig, args, arg0_type, id);
+		
 	}
 	case SN_op_Equality:
 	case SN_op_Inequality:
-		g_assert (fsig->param_count == 2 && fsig->ret->type == MONO_TYPE_BOOLEAN &&
-				  mono_metadata_type_equal (fsig->params [0], type) &&
-				  mono_metadata_type_equal (fsig->params [1], type));
+		if (fsig->param_count != 2 )
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1257]: Ditto

File: src/mono/mono/mini/simd-intrinsics.c [hunk 6]
```diff
@@ -1197,27 +1240,32 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
...
+			return NULL;
 		switch (id) {
 			case SN_op_Equality: return emit_xequal (cfg, klass, args [0], args [1]);
 			case SN_op_Inequality: return emit_not_xequal (cfg, klass, args [0], args [1]);
 			default: g_assert_not_reached ();
 		}
+	case SN_op_OnesComplement:
+	case SN_op_UnaryNegation:
+		if (fsig->param_count != 1 )
```
Comment (fanyang-mono) [inline src/mono/mono/mini/simd-intrinsics.c:1266]: Ditto

**Comment by fanyang-mono** [comment]
The failures on CI are irrelevant to this PR.
